inc/namespace.php

```diff
@@ -424,3 +436,70 @@ function remove_xray_metadata( array $metadata ) : array {
 
 	return $metadata;
 }
+
+/**
+ * Return an AWS CloudFront Client instance.
+ *
+ * @return \Aws\CloudFront\CloudFrontClient
+ */
+function get_cloudfront_client() : CloudFrontClient {
+	return get_aws_sdk()->createCloudFront( [
+		'version' => '2019-03-26',
+	] );
+}
+
+/**
+ * Create purge request to invalidate CDN cache.
+ *
+ * @param array $paths_patterns A list of the paths that you want to invalidate.
+ *                              The path is relative to the CDN host, A leading / is optional.
+ *                              e.g  for http://altis-dxp.com/images/image2.jpg
+ *                              specify images/image2.jpg or /images/image2.jpg
+ *
+ *                              You can also invalidate multiple files simultaneously by using the * wildcard.
+ *                              The *, which replaces 0 or more characters, must be the last character in the invalidation path.
+ *                              e.g /images/* - will invalidate all files in a directory
+ *
+ * @return bool Returns true if invalidation successfully created, false on failure.
+ */
+function purge_cdn_paths( array $paths_patterns ) : bool {
+	$client = get_cloudfront_client();
+
+	$distribution_id = '';
+
+	if ( defined( 'CLOUDFRONT_DISTRIBUTION_ID' ) ) {
+		$distribution_id = CLOUDFRONT_DISTRIBUTION_ID;
+	}
+
+	/**
+	 * Filters the CloudFront Distribution ID used when purging CDN paths.
+	 *
+	 * @param string $distribution_id The ID to set.
+	 */
+	$distribution_id = apply_filters( 'altis.cloud.cdn_distribution_id', $distribution_id );
```

Comment:
@joehoyle I'd recommend adding something to the docs about this filter, if it hasn't already. If you add `get_aws_sdk` from https://github.com/humanmade/altis-core/blob/master/docs/README.md#aws-sdk this will fatal locally.

Reply:
Hmm yeah, you need to make sure any code calling this is wrapped in checks that it's running in Cloud